src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,321 @@
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface LineChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    connectNulls?: boolean,
+    domainMin?: number | "Auto"
+}
+
+export interface BarChartOptions {
+    barColor?: ColorDefinition | ColorDefinition[],
+    thresholds?: BarChartThreshold[]
+}
+
+export interface BarChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdBarColor?: ColorDefinition
+}
+
+export interface AreaChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    areaColor?: ColorDefinition | ColorDefinition[],
+}
+
+export default function TimeSeriesChart(props: TimeSeriesChartProps) {
+    let layoutContext = useContext(LayoutContext);
+    let intervalType = props.intervalType || "Month";
+
+    const DayTick = ({ x, y, stroke, payload }: any) => {
+        var value = payload.value;
+        let currentDate = new Date(value);
+        if (intervalType == "Month") {
+            return <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 15} textAnchor="middle" fontSize="12">{currentDate.getDate()}</text>;
+        } else if (intervalType == "SixMonth" ){
+            let monthLabel = currentDate.getDate() === 1 ? format(currentDate, "LLL") : "";
+            let dayLabel = currentDate.getDate().toString();
+            return <>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 8} textAnchor="middle" fontSize="11">{monthLabel}</text>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 24} textAnchor="middle" fontSize="12">{dayLabel}</text>
+            </>;
+        } else if (intervalType == "Week" ){
+            let dayOfWeek: string = "";
+            for (let i = 0; i < 7; i++) {
+                if (currentDate.getTime() == value) {
+                    dayOfWeek = format(currentDate, "EEEEEE");
+                    break;
+                }
+                currentDate = add(currentDate, { days: 1 });
+            }
+            return <>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 8} textAnchor="middle" fontSize="11">{dayOfWeek}</text>
+                <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 24} textAnchor="middle" fontSize="12">{currentDate.getDate()}</text>
+            </>;
+        } else if ( intervalType == "Day"){
+            const startTime = new Date(props.intervalStart);
+            startTime.setHours(0, 0, 0, 0);
+            return <>
+                {currentDate.getTime() === startTime.getTime() && <text className={isToday(currentDate) ? "today" : ""} fill="var(--mdhui-text-color-2)" x={x} y={y + 8} textAnchor="middle" fontSize="11">{format(currentDate, "EEEEEE")}</text>}
```

Comment:
Is including this intentional?  Looks like there's a random day of week at the beginning:
<img width="95" alt="Screenshot 2024-05-03 at 2 35 23 PM" src="https://github.com/CareEvolution/MyDataHelpsUI/assets/1643171/81f2d580-17dd-425c-b21b-b9e025358ba7">

Also looks like there's some random bolding of the times

Reply:
On the "Line Chart Intraday" story

Reply:
It was intentional, but I can take it or leave it.  I was trying to provide a bit of "context" for the clock time being shown; it's perhaps unnecessary as that information will also (likely) be displayed in the DateRangeCoordinator.  I kind of put it there because the DateRangeCoordinator isn't *required* to provide context, in which case the data being displayed may be more confusing (though "Friday" in this case doesn't help much, where a full date may be better).  If it seems redundant, I'm fine to pull it out.

The bolding is... probably a holdover from the code that attempts to bold the labels on the X-Axis if the label is for "today" - should probably just pull that styling out in the intraday layout as it won't be providing any clarity.

Reply:
Yeah i'd definitely take it out and make the styling non-bold for the intraday use case; I don't think the context would be useful since it'll either have an explicit label or be in the date range coordinator etc.